jobs/rocoto/prep.sh

```diff
@@ -1,129 +1,141 @@
 #! /usr/bin/env bash
 
-source "$HOMEgfs/ush/preamble.sh"
+source "${HOMEgfs}/ush/preamble.sh"
 
 ###############################################################
 # Source FV3GFS workflow modules
-. $HOMEgfs/ush/load_fv3gfs_modules.sh
+. ${HOMEgfs}/ush/load_fv3gfs_modules.sh
 status=$?
-[[ $status -ne 0 ]] && exit $status
+[[ ${status} -ne 0 ]] && exit ${status}
 
 ###############################################################
 # Source relevant configs
 configs="base prep"
-for config in $configs; do
-    . $EXPDIR/config.${config}
+for config in ${configs}; do
+    . ${EXPDIR}/config.${config}
     status=$?
-    [[ $status -ne 0 ]] && exit $status
+    [[ ${status} -ne 0 ]] && exit ${status}
 done
 
 ###############################################################
 # Source machine runtime environment
-. $BASE_ENV/${machine}.env prep
+. ${BASE_ENV}/${machine}.env prep
 status=$?
-[[ $status -ne 0 ]] && exit $status
+[[ ${status} -ne 0 ]] && exit ${status}
 
 ###############################################################
 # Set script and dependency variables
 export COMPONENT=${COMPONENT:-atmos}
 export OPREFIX="${CDUMP}.t${cyc}z."
-export COMOUT="$ROTDIR/$CDUMP.$PDY/$cyc/$COMPONENT"
+export COMOUT="${ROTDIR}/${CDUMP}.${PDY}/${cyc}/${COMPONENT}"
 export MAKE_PREPBUFR=${MAKE_PREPBUFR:-"YES"}
-[[ ! -d $COMOUT ]] && mkdir -p $COMOUT
+[[ ! -d ${COMOUT} ]] && mkdir -p ${COMOUT}
+[[ ! -d ${COMIN_OBS} ]] && mkdir -p ${COMIN_OBS}
 
 ###############################################################
 # If ROTDIR_DUMP=YES, copy dump files to rotdir
-if [ $ROTDIR_DUMP = "YES" ]; then
-   $HOMEgfs/ush/getdump.sh $CDATE $CDUMP $DMPDIR/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/${COMPONENT} $COMOUT
+if [[ ${ROTDIR_DUMP} = "YES" ]]; then
+   ${HOMEgfs}/ush/getdump.sh "${CDATE}" "${CDUMP}" "${DMPDIR}/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/${COMPONENT}" "${COMIN_OBS}"
```

Comment:
```suggestion
   ${HOMEgfs}/ush/getdump.sh "${CDATE}" "${CDUMP}" "${DMPDIR}/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/atmos" "${COMIN_OBS}"
```
obsproc puts this in `atmos`.  Lets use that verbatim.